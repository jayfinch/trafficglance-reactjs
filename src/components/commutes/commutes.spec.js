import React from 'react'
import Commutes from './commutes'
import renderer from 'react-test-renderer'

describe('<Commutes />', () => {
  describe('render', () => {
    it('Should render without config', () => {
      let tree = renderer.create(
        <Commutes
          commutes={[]}
          distanceUnit='mi'
          onClickRefresh={jest.fn()}
          configError
        />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })

    it('Should render with commutes', () => {
      let tree = renderer.create(
        <Commutes
          commutes={[
            {
              id: 0,
              name: 'foo',
              url: 'bar',
              fetchingTraffic: false
            }
          ]}
          distanceUnit='mi'
          onClickRefresh={jest.fn()}
        />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
