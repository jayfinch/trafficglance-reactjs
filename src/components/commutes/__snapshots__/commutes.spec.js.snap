exports[`<Commutes /> render Should render with commutes 1`] = `
<div
  className="row"
  id="routes">
  <div
    className="route col-sm-6 col-md-4">
    <div
      className="card">
      <div
        className="card__name">
        <a
          href="bar"
          target="_blank">
          foo
        </a>
        
      </div>
      <div
        className="row">
        <div
          className="col-xs-5 col-sm-4 chart">
          <button
            className="card__fetch-button"
            onClick={[Function]}
            title="Refresh">
            <img
              className="card__refresh-btn img-responsive center-block"
              src="images/refresh-circle.svg" />
          </button>
        </div>
        <div
          className="col-xs-7 col-sm-8 card-stats">
          <div
            className="card-stats__duration">
            –––
          </div>
          <div
            className="card-stats__detail">
            –––
          </div>
          <div
            className="card-stats__detail">
            –––
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Commutes /> render Should render without config 1`] = `
<div
  className="row"
  id="routes">
  <div
    className="loading">
    You must set up your config
  </div>
</div>
`;
